### ISSUE TITLE
collate_fn in train.py doesn't apply changes in presets, causing malformed batches

### ISSUE THREAD

**Issue by engiecat** [r9y9/wavenet_vocoder#63]
Hi. 
First, thank you for providing such great repos :D

While testing with multispeaker dataset, i just discovered that `collate_fn` is not affected by hparams.parse_json in train.py, resulting in hparams changes NOT applied while creating batches
(This was checked by putting `print(hparams_debug_string())` in `collate_fn` and run training while applying different presets)

This has some consequences.
For example, when trying to train multispeaker dataset, the `gin_channels` in collate_fn is fixed to -1 (even if it is changed to 16 in preset json file), 
https://github.com/r9y9/wavenet_vocoder/blob/4116d34eddd3f841a905660d761d186b483f1846/train.py#L368

causing `global_conditioning` to be False (incorrectly)
This causes `g_batch` to be None, and this causes `sanity_check` to fail.
https://github.com/r9y9/wavenet_vocoder/blob/4116d34eddd3f841a905660d761d186b483f1846/train.py#L73-L76

Same thing can occur when `cin_channels, max_time_sec, sample_rate, upsample_conditional_features, input_type, quantize_channels` are changed via preset JSON files. 

This is quite similar with https://github.com/r9y9/deepvoice3_pytorch/issues/80

In short, this works when using ljspeech preset(as it is identical with hparams.py), but doesn't work with other presets

**Comment by engiecat** [comment]
So, this works when using ljspeech preset(as it is identical with `hparams.py`), but doesn't work well with other presets

**Comment by engiecat** [comment]
i will also try to find out why :/ probably ProcessPoolExecutor works different in windows from linux.
